pkg/cimfs/cim_writer_windows.go

```diff
@@ -62,6 +62,33 @@ func Create(imagePath string, oldFSName string, newFSName string) (_ *CimFsWrite
 	return &CimFsWriter{handle: handle, name: filepath.Join(imagePath, fsName)}, nil
 }
 
+// CreateMergedCim creates a new merged CIM from the CIMs provided in the `cimPaths` array.  CIM at index 0 is
+// considered to be the topmost CIM and the CIM at index `cimPath.length-1` is considered the base CIM
+// on. When mounting this merged CIM the source CIMs MUST be provided in the same order. Ensure all CIM paths
+// are absolute paths. `imageDir` is the directory inside which the merged cim is stored with the name `name`.
+func CreateMergedCim(imageDir, name string, cimPaths []string) (err error) {
+	if !IsMergedCimSupported() {
+		return fmt.Errorf("merged CIMs are not supported on this version")
+	}
+	cim, err := Create(imageDir, "", name)
+	if err != nil {
+		return fmt.Errorf("create cim image: %w", err)
+	}
+	defer winapi.CimCloseImage(cim.handle)
+
+	for _, lpath := range cimPaths {
+		if err := winapi.CimAddFsToMergedImage(cim.handle, lpath); err != nil {
```

Comment:
What exactly does this do?

Reply:
Basically I don't undestand what this step of adding merges to a cim file is for. I thought merging was purely a mount-time operation.

Reply:
There are two types of merged CIMs. On disk merge and in memory merge. What you are referring to is an in memory merge. However, with that we can't reuse an already merged CIM with multiple containers (every mount operation will create the merge). With an on disk merge, we create a merged CIM once (which basically just a CIM that contains metadata about underlying layer CIMs) and mount that merged CIM every time we start a container. `CimAddFsToMergedImage` call adds a CIM into such a merged CIM. Does that help?